controllers/hivelocitymachine_controller.go

```diff
@@ -120,8 +121,10 @@ func (r *HivelocityMachineReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	secretManager := secretutil.NewSecretManager(log, r.Client, r.APIReader)
 	hvAPIKey, _, err := getAndValidateHivelocityAPIKey(ctx, req.Namespace, hvCluster, secretManager)
 	if err != nil {
+		conditions.MarkFalse(hvCluster, infrav1.CredentialsAvailableCondition, infrav1.HivelocityWrongAPIKeyReason, clusterv1.ConditionSeverityError, err.Error())
 		return hvAPIKeyErrorResult(ctx, err, hivelocityMachine, infrav1.DeviceReadyCondition, r.Client)
 	}
+	conditions.MarkTrue(hvCluster, infrav1.CredentialsAvailableCondition)
```

Comment:
this is too early to set it on true right? You set it a second time to true in device.go btw...

Reply:
right, first we check the API key the SSH key and if we found both then make true